src/number_theory/number_field.lean

```diff
@@ -142,3 +144,89 @@ noncomputable def ring_of_integers_equiv : ring_of_integers ℚ ≃+* ℤ :=
 ring_of_integers.equiv ℤ
 
 end rat
+
+namespace adjoin_root
+
+section
+
+open_locale polynomial
+
+local attribute [-instance] algebra_rat
+
+/-- The quotient of `ℚ[X]` by the ideal generated by an irreducible polynomial of `ℚ[X]`
+is a number field. -/
+instance {f : ℚ[X]} [hf : irreducible f] : number_field (adjoin_root f) :=
```

Comment:
@ericrbg you may be interested in this use of `irreducible f` if you are still planning to make it a `def`.

Reply:
I replaced all these with `fact`s, `adjoin_root` is the only place that really uses it. #14713